File: src/main/java/eterea/api/rest/model/Usuario.java
/**
 * 
 */
package eterea.api.rest.model;

import java.io.Serializable;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table(name = "usuarios")
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class Usuario extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 3577203074453281119L;

	@Id
	@Column(name = "nombre")
	private String login;

	@Column(name = "usu_descripcion")
	private String descripcion;

	@Column(name = "usu_password")
	private String password;

	@Column(name = "usu_correo")
	private String email;

	@Column(name = "usu_nivel")
	private Integer nivel;

	@Column(name = "usu_pin")
	private String pin;

	@Column(name = "usu_cierrectype")
	private String cierreRecipientType;

	@Column(name = "usu_consrectype")
	private String consolidadoRecipientType;

	@Column(name = "clave")
	private Long usuarioId;
	
}

File: src/main/java/eterea/core/api/rest/kotlin/model/Usuario.kt
package eterea.core.api.rest.kotlin.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "usuarios")
data class Usuario(

    @Id
    @Column(name = "nombre")
    var login: String? = null,

    @Column(name = "Usu_Descripcion")
    var descripcion: String? = null,

    @Column(name = "Usu_Password")
    var password: String? = null,

    @Column(name = "Usu_Correo")
    var email: String? = null,

    @Column(name = "Usu_Nivel")
    var nivel: Int = 0,

    @Column(name = "Usu_PIN")
    var pin: String? = null,

    @Column(name = "Usu_CierRecType")
    var cierreRecipientType: String? = null,

    @Column(name = "Usu_ConsRecType")
    var consolidadoRecipientType: String? = null,

    @Column(name = "clave")
    var usuarioId: Long? = null

) : Auditable() {

    data class Builder(
        private var login: String? = null,
        private var descripcion: String? = null,
        private var password: String? = null,
        private var email: String? = null,
        private var nivel: Int = 0,
        private var pin: String? = null,
        private var cierreRecipientType: String? = null,
        private var consolidadoRecipientType: String? = null,
        private var usuarioId: Long? = null
    ) {
        fun login(login: String) = apply { this.login = login }
        fun descripcion(descripcion: String) = apply { this.descripcion = descripcion }
        fun password(password: String) = apply { this.password = password }
        fun email(email: String) = apply { this.email = email }
        fun nivel(nivel: Int) = apply { this.nivel = nivel }
        fun pin(pin: String) = apply { this.pin = pin }
        fun cierreRecipientType(cierreRecipientType: String) = apply { this.cierreRecipientType = cierreRecipientType }
        fun consolidadoRecipientType(consolidadoRecipientType: String) = apply { this.consolidadoRecipientType = consolidadoRecipientType }
        fun usuarioId(usuarioId: Long) = apply { this.usuarioId = usuarioId }

        fun build() = Usuario(
            login,
            descripcion,
            password,
            email,
            nivel,
            pin,
            cierreRecipientType,
            consolidadoRecipientType,
            usuarioId
        )
    }
}
